IDEA-277826 Do not create test configurations when no test sources found

karimai-gg opened the pull request:
When creating configurations for AllInDirectoryGradleConfigurationProducer, take into consideration the
fact of actual test sources being available within the given context
location.

karimai-gg commented:
With regards to tests : I tried running GradleTestRunConfigurationProducerTest locally but all the tests are failing (run failure ) so I am not sure if there is something that need to be set up to get the tests to run and pass so that I can a test for my change if necessary !

nskvortsov reviewed (changes requested):
A test is required. It should display this change does whats intended.

Some tweaks to default JUnit run configurations are required to run tests. Please see the https://github.com/JetBrains/intellij-community/blob/master/README.md for details

karimai-gg commented:
I see that AllInDirectoryGradleConfigurationProducer and the remaining producers have been changed since last time and that we no longer use the sourceFolders when checking if a configuration can be set up from a given context ! 
I will test it and see if https://youtrack.jetbrains.com/issue/IDEA-277826 is also fixed with these new changes and update the PR accordingly.

DeveloperHacker commented:
This fix break a `com.android.tools.idea.testartifacts.gradle.AndroidGradleConfigurationProducersTest#testCanCreateGradleConfigurationFromTestDirectory`. Please wait for my small revert for this, and I investigate why this test fails.

karimai-gg commented:
alright, please let me know once everything lands on your side !

DeveloperHacker commented:
So, `com.android.tools.idea.testartifacts.gradle.AndroidGradleConfigurationProducersTest#testCanCreateGradleConfigurationFromTestDirectory` test fails because it expected that configuration will be created if we don't have relevant test tasks. I check it on previous versions. Idea shows balloon 'test tasks not found' when I try to run run configuration from app/../java directory. Also, I find that this test was created by you. Could you fix it so that I can apply fix which removes this notification?
Also, could you see on `com.android.tools.idea.testartifacts.gradle.AndroidGradleConfigurationProducersTest#testCanCreateGradleConfigurationFromTestDirectoryKotlin`?

karimai-gg commented:
I am attempting to submit a fix in our codebase to disable the failing tests for the moment. Once it lands, I will let you know and you can cherry-pick into your codebase !

Please let me know if you do not have access to the mirroring repo ?

nskvortsov commented:
it seems like this review got push-force and now contains the same changes, as https://github.com/JetBrains/intellij-community/pull/1734/

Is it intended?

karimai-gg commented:
This was unintentional and I didn't notice it ! Reverted. Thank you.

I have also reverted my change to AllInDirectoryGradleConfigurationProducer which seem to be not applicable after the recent changes made by @DeveloperHacker ! I only left the test for now.

Also @DeveloperHacker  I have submitted a change that fixxes the test breakages that you can cherry pick from here : https://android.googlesource.com/platform/tools/adt/idea/+/e61448032e6997cc931bf8bbaac7d96400d5d216

Please let me know if it fixes the issue you had.

karimai-gg commented:
I have updated the PR to be rebased on the recent changes made by your team to the Gradle Run configuration producers. The test I added passes now and I hope that it also highlights what does my PR do ! 

Please let me know if this looks right to you ! what I basically added was that : if we create a RunConfiguration and this doesn't have any tasks in it, then we should not attempt to create it because nothing will be executed in the end, and we will get 'No tasks Found'

karimai-gg commented:
Hello,
Just checking in if there are any updates on this ? 

Thanks.

DeveloperHacker commented:
@karimai-gg Andrei is pulling changes from android studio repo, and he said that he will pull your changes after this. cc @andrei-kuznetsov

andrei-kuznetsov commented:
True. I'll pull this change after I finish merge of Android Studio into our master. Mid of next week, probably

andrei-kuznetsov commented:
@karimai-gg, running tests right now. Sorry for the delay.
